stories/core/Table.stories.tsx

```diff
@@ -718,3 +759,119 @@ export const NoData: Story<TableProps> = ({ columns, data, ...rest }) => {
   data: [],
 };
+
+export const ControlledState: Story<TableProps> = (args) => {
+  const { columns, data, ...rest } = args;
+
+  const [selectedRows, setSelectedRows] = useState<Record<string, boolean>>({});
+
+  const onSelect = useCallback((rows, state) => {
+    action(
+      `Selected rows: ${JSON.stringify(rows)}, Table state: ${JSON.stringify(
+        state,
+      )}`,
+    )();
+  }, []);
+
+  const tableColumns = useMemo(
+    () => [
+      {
+        Header: 'Table',
+        columns: [
+          {
+            id: 'name',
+            Header: 'Name',
+            accessor: 'name',
+          },
+          {
+            id: 'description',
+            Header: 'Description',
+            accessor: 'description',
+          },
+        ],
+      },
+    ],
+    [],
+  );
+
+  const tableData = useMemo(
+    () => [
+      { name: 'Name1', description: 'Description1' },
+      { name: 'Name2', description: 'Description2' },
+      { name: 'Name3', description: 'Description3' },
+    ],
+    [],
+  );
+
+  const controlledState = useCallback(
+    (state) => {
+      return {
+        ...state,
+        selectedRowIds: { ...selectedRows },
+      };
+    },
+    [selectedRows],
+  );
+
+  // When using `useControlledState` we are fully responsible for the state part we are modifying.
+  // Therefore we want to keep our outside state (`selectedRows`) in sync with inside table state (`state.selectedRowIds`).
+  const tableStateReducer = (
+    newState: TableState,
+    action: ActionType,
+  ): TableState => {
+    switch (action.type) {
+      case actions.toggleRowSelected: {
+        const newSelectedRows = {
+          ...selectedRows,
+        };
+        if (action.value) {
+          newSelectedRows[action.id] = true;
+        } else {
+          delete newSelectedRows[action.id];
+        }
+        setSelectedRows(newSelectedRows);
+        newState.selectedRowIds = newSelectedRows;
+        break;
+      }
+      default:
+        break;
+    }
+    return newState;
+  };
+
+  return (
+    <>
+      <InputGroup label='Control selected rows' style={{ marginBottom: 11 }}>
+        {tableData.map((data, index) => (
+          <Checkbox
+            key={index}
+            label={data.name}
+            checked={selectedRows[index]}
+            onChange={(e) => {
+              setSelectedRows((rowIds) => {
+                const selectedRowIds = { ...rowIds };
+                if (e.target.checked) {
+                  selectedRowIds[index] = true;
+                } else {
+                  delete selectedRowIds[index];
+                }
+                return selectedRowIds;
+              });
+            }}
+          />
+        ))}
+      </InputGroup>
+      <Table
+        columns={columns || tableColumns}
+        data={data || tableData}
+        emptyTableContent='No data.'
+        onSelect={onSelect}
+        useControlledState={controlledState}
+        stateReducer={tableStateReducer}
+        {...rest}
```

Comment:
Add `isSelectable` here to make the story more "complete" without args.

Reply:
Added.